### PR TITLE
Add support for running docker-ce from a snap (Fixes: #9)

### PR DESCRIPTION
**Pull request by superm1**
This adjusts the following:
* Systemd unit dependencies
* Location of docker daemon (/usr/bin/ -> /snap/bin)
* Default data storage location
* Configuration storage location

It allows optionally using both 'jq' and 'docker' from the snaps
provided by the Ubuntu server 18.04 installation process.

### CODE REVIEW

File: hassio_install.sh [hunk 1]
```diff
@@ -21,7 +24,17 @@ command -v dbus-daemon > /dev/null 2>&1 || { echo "[Error] Please install dbus f
 command -v apparmor_parser > /dev/null 2>&1 || echo "[Warning] No AppArmor support on host."
 command -v nmcli > /dev/null 2>&1 || echo "[Warning] No NetworkManager support on host."
 
+#detect if running on snapped docker
+if snap list docker >/dev/null 2>&1; then
+    SNAP=1
```
Comment (pvizeli) [inline hassio_install.sh]: `SNAP=true`

File: hassio_install.sh [hunk 2]
```diff
@@ -2,8 +2,11 @@
 set -e
 
 ARCH=$(uname -m)
+PREFIX=""
```
Comment (pvizeli) [inline hassio_install.sh]: A default:
`SNAP=false`

File: hassio_install.sh (hunk 1 above)
Comment (superm1) [reply hassio_install.sh]: Adjusted, thanks

**Comment by superm1** [comment]
>Why you add an x ?

I've seen this uses in autotools (`configure.ac`) and bash scripts in the past.  I suppose it's related to potentially having a bad expansion that includes something like a backspace character?  If you would prefer I remove it, i'll remove it.

**Comment by pvizeli** [comment]
I don't think with quota we run into a problem. Please remove it, so we use the same style over all files. Thanks

**Comment by superm1** [comment]
>  Please remove it, so we use the same style over all files. Thanks

OK adjusted, thanks.

**Pull request merged**